File: lab/python/primos/primos_ate_n.py
def eh_primo(n):
    if n < 2:
        return False

    for x in range(2, int(n**0.5) + 1):
        if n % x == 0:
            return False
    return True


def primos_ate(n):
    for divisor in range(2, n+1):
        if eh_primo(divisor):
            print(divisor, end=" ")
    print()


primos_ate(10)

File: lab/c/primos/primos_ate_n.c
#include <stdio.h>
#include <math.h>


int eh_primo(int n);
void primos_ate(int n);


int main(){
    primos_ate(10);
}


int eh_primo(int n){
    if (n < 2){
        return 0;
    }

    int raiz = (int) pow(n, 0.5); // raiz quadrada truncada

    for (int x=2; x < raiz + 1; x++){
        if (n % x == 0){
            return 0;
        }
    }
    return 1;
}


void primos_ate(int n){
    for (int divisor=2; divisor <= n; divisor++){
        if (eh_primo(divisor)){
            printf("%d ", divisor);
        }
    }
    printf("\n");
}
